runtime/compiler/aarch64/runtime/PicBuilder.spp

```diff
@@ -235,27 +243,157 @@ const_jitResolveStaticMethod:
 const_jitResolveSpecialMethod:
 	.dword	jitResolveSpecialMethod
 
+// Handles calls to unresolved data snippets
+//
+// in:		x3 = resolve helper address
+//		x30 = snippet data
+// out:
+// trash:
+
+// For unresolved data access, we generate following instructions
+//  b     UnresolvedDataSnippet ; change this to "movz x9, #imm"
+//  movk  x9, #0, LSL #16
+//  movk  x9, #0, LSL #32
+//  movk  x9, #0, LSL #48
+//  use x9 for data access
```

Comment:
I realize this code was inspired from the ARM implementation where it uses a hard-coded `gr3` in the mainline code to hold the resolved address, but even in that world, how does it actually reserve `gr3` for this purpose?  I would think every unresolved memory reference on every instruction would need a register dependency on `gr3` to prevent `gr3` from being assigned to a virtual register (and hence be clobbered by the resolution code).  I can't see how that is done.  You'll have a similar problem with `r9` on AArch64.

Reply:
Actually, the ARM implementation uses `add` instructions (with shifted offset) to build up the address using the base register of the memory reference (whatever that base register happens to be).  If you are using `x9` then my question still remains on how you plan on ensuring that register will be free.